rplugin/python3/LanguageClient/RPC.py

```diff
@@ -60,13 +60,25 @@ def notify(self, method: str, params: Dict[str, Any]) -> None:
     def serve(self):
         content_length = 0
         while not self.infile.closed:
-            line = self.infile.readline().decode("UTF-8").strip()
+            try:
+                data = self.infile.readline()
+                line = data.decode("UTF-8").strip()
+            except Exception:
+                msg = "Failed to decode message as UTF-8: " + data
+                logger.exception(msg)
+                continue
             if line:
                 header, value = line.split(":")
                 if header == "Content-Length":
                     content_length = int(value)
             else:
-                content = self.infile.read(content_length).decode("UTF-8")
+                try:
+                    data = self.infile.read(content_length)
+                    content = data.decode("UTF-8")
+                except Exception:
```

Comment:
Instead of catching all exception here, would it be better catching `UnicodeError`?